WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Customs/ItemDetails/ShippingLabelCustomsFormItemDetailsViewModel.swift

```diff
@@ -54,3 +74,38 @@ final class ShippingLabelCustomsFormItemDetailsViewModel: ObservableObject {
         self.originCountry = countries.first(where: { $0.code == item.originCountry }) ?? Country(code: "", name: "", states: [])
     }
 }
+
+// MARK: - Validation
+//
+extension ShippingLabelCustomsFormItemDetailsViewModel {
+    var hasValidDescription: Bool {
+        description.trimmingCharacters(in: .whitespacesAndNewlines).isNotEmpty
+    }
+
+    var validatedValue: Double? {
+        guard let numericValue = Double(value), numericValue > 0 else {
+            return nil
+        }
+        return numericValue
+    }
+
+    var validatedWeight: Double? {
+        guard let numericWeight = Double(weight), numericWeight > 0 else {
+            return nil
+        }
+        return numericWeight
+    }
+
+    var hasValidOriginCountry: Bool {
+        originCountry.code.isNotEmpty
+    }
+
+    var hasValidHSTariffNumber: Bool {
+        if hsTariffNumber.isNotEmpty,
+           (hsTariffNumber.count != 6 ||
+                hsTariffNumber.filter({ "0"..."9" ~= $0 }).count != 6) {
```

Comment:
I noticed that I can enter an HS tariff number over 6 digits long, and I get the validation error. I found this a little confusing because our docs say this (which made me think that at least 6 digits were needed but I wasn't sure why longer numbers would be invalid):

> Though some codes may be longer, you only need to include the first six digits.

On Android and web, we limit this field to 6 digits so it isn't possible to enter anything longer. What do you think of matching this behavior (limit the text field entry to 6 digits)?

Reply:
Thank you for pointing that out! I have implemented the behavior in d6b94761968b6de4dae5d33a587b4673e0aae996.